src/libsync/bulkpropagatorjob.cpp

```diff
@@ -0,0 +1,869 @@
+            return;
+        }
+        qCWarning(lcBulkPropagatorJob()) << item->_file << item->_renameTarget;
+        fileToUpload._file = item->_file = item->_renameTarget;
+        fileToUpload._path = propagator()->fullLocalPath(fileToUpload._file);
+        item->_modtime = FileSystem::getModTime(newFilePathAbsolute);
+    }
+
+    QString path = fileToUpload._file;
+
+    qCInfo(lcBulkPropagatorJob) << propagator()->fullRemotePath(path) << "transmission checksum" << transmissionChecksumHeader;
+    currentHeaders[checkSumHeaderC] = transmissionChecksumHeader;
+
+    const QString fileName = fileToUpload._path;
+
+    UploadFileParameters newUploadFile{propagator()->account(), item, fileToUpload,
+                propagator()->fullRemotePath(path), fileName,
+                fileSize, currentHeaders};
+
+    _uploadFileParameters.push_back(std::move(newUploadFile));
+
+    if (_items.empty() && _checksumsJobs.empty()) {
+        triggerUpload();
+    }
+}
+
+void BulkPropagatorJob::triggerUpload()
+{
+    auto uploadParametersData = std::vector<OneUploadFileData>{};
+    uploadParametersData.reserve(_uploadFileParameters.size());
+
+    for(auto &oneFile : _uploadFileParameters) {
+        // job takes ownership of device via a QScopedPointer. Job deletes itself when finishing
+        auto device = std::make_unique<UploadDevice>(
+                oneFile._localPath, 0, oneFile._fileSize, &propagator()->_bandwidthManager);
+        if (!device->open(QIODevice::ReadOnly)) {
+            qCWarning(lcBulkPropagatorJob) << "Could not prepare upload device: " << device->errorString();
+
+            // If the file is currently locked, we want to retry the sync
+            // when it becomes available again.
+            if (FileSystem::isFileLocked(oneFile._localPath)) {
+                emit propagator()->seenLockedFile(oneFile._localPath);
+            }
+            // Soft error because this is likely caused by the user modifying his files while syncing
+            abortWithError(oneFile._item, SyncFileItem::SoftError, device->errorString());
+            return;
+        }
+        oneFile._headers["OC-Path"] = oneFile._remotePath.toUtf8();
+        uploadParametersData.push_back({std::move(device), oneFile._headers});
+    }
+
+    auto job = std::make_unique<PutMultiFileJob>(propagator()->account(), QUrl{}, std::move(uploadParametersData), this);
+    connect(job.get(), &PutMultiFileJob::finishedSignal, this, &BulkPropagatorJob::slotPutFinished);
+    connect(job.get(), &PutMultiFileJob::uploadProgress, this, &BulkPropagatorJob::slotUploadProgress);
+    //connect(job.get(), &PutMultiFileJob::uploadProgress, device.get(), &UploadDevice::slotJobUploadProgress);
+    connect(job.get(), &QObject::destroyed, this, &BulkPropagatorJob::slotJobDestroyed);
+    //adjustLastJobTimeout(job.get(), oneFile._fileSize);
+    auto jobCopy = job.get();
+    _jobs.append(job.release());
+    jobCopy->start();
```

Comment:
Why not:
```cpp
_jobs.append(job.get());
job.release()->start();
```
or
```cpp
_jobs.append(job.get());
job->start();
job->release();
```

Reply:
but the second one will fail to pass CI because we ignore the result of the call to `job->release()`
and somehow that rule is making sense
so I have to go with the first version